torch/_tensor.py

```diff
@@ -209,8 +209,18 @@ def __deepcopy__(self, memo):
             return new_tensor
 
     def __reduce_ex__(self, proto):
+        skip_data = getattr(torch.serialization._serialization_tls, "skip_data", False)
+        materialize_fake_tensors = getattr(
+            torch.serialization._serialization_tls, "materialize_fake_tensors", False
+        )
         state = torch._utils._get_obj_state(self)
-        if type(self) is Tensor and not state:
+        # Ignore all state when using FakeTensor with skip_data(materialize_fake_tensors) because FakeTensor has
```

Comment:
I think this is sub-optimal in the long term. We should have a clearer contract on how our FakeTensor -> real Tensor with no data.
Dropping every single field from it might not be acceptable for everyone and might be interesting if there was a method to extract such Tensor from it easily.

This sounds ok as a temporary solution here I think

Reply:
I agree, technically what we want is to drop every attribute that is in the __dict__ of a "normal" FakeTensor, but keep anything else, is that right?

But agreed that for now it seems tricky to handle this because I'm not sure what is in a "norma" FakeTensor's __dict__ is stable